src/executePrettier/buildPrettierOptions.test.js

```diff
@@ -109,40 +109,60 @@ it('uses vue as the parser if current scope is listed as a Vue SFC scope in sett
+  });
 });
 
-it('overrides values with editorconfig values if editor config is enabled', () => {
-  const editor = buildMockEditor();
-  const fakePrettierOptions = { tabWidth: 2, printWidth: 80, useTabs: false };
-  const fakeEditorConfigOptions = { tabWidth: 4, printWidth: 100, useTabs: true };
-  getPrettierOptions.mockImplementation(() => fakePrettierOptions);
-  buildEditorConfigOptions.mockImplementation(() => fakeEditorConfigOptions);
-  shouldUseEditorConfig.mockImplementation(() => true);
-  getCurrentFilePath.mockImplementation(() => 'parent-dir/foo.js');
-
-  const actual = buildPrettierOptions(editor);
-
-  expect(actual).toEqual(fakeEditorConfigOptions);
-});
-
-it('overrides values with prettier config values if one exists', () => {
-  const editor = buildMockEditor();
-  const fakePrettierOptions = { tabWidth: 2, printWidth: 80, useTabs: false };
-  const fakePrettierConfigOptions = { tabWidth: 4, printWidth: 100, useTabs: true };
-  const mockPrettierInstance = { resolveConfig: { sync: jest.fn(() => fakePrettierConfigOptions) } };
-  getPrettierInstance.mockImplementation(() => mockPrettierInstance);
-  getCurrentFilePath.mockImplementation(() => 'parent-dir/foo.js');
-  getPrettierOptions.mockImplementation(() => fakePrettierOptions);
-
-  const actual = buildPrettierOptions(editor);
-
-  expect(actual).toEqual(fakePrettierConfigOptions);
+describe('when a prettier config is not found', () => {
+  it('uses settings from prettier-atom and Editor Config when enabled', () => {
+    const editor = buildMockEditor();
+    const fakePrettierOptions = { singleQuote: true, tabWidth: 2 };
+    const fakeEditorConfigOptions = { tabWidth: 4 };
+    const mockPrettierInstance = { resolveConfig: { sync: jest.fn(() => null) } };
+    getPrettierInstance.mockImplementation(() => mockPrettierInstance);
+    buildEditorConfigOptions.mockImplementation(() => fakeEditorConfigOptions);
+    shouldUseEditorConfig.mockImplementation(() => true);
+    getCurrentFilePath.mockImplementation(() => 'parent-dir/foo.js');
+    getPrettierOptions.mockImplementation(() => fakePrettierOptions);
+
+    const actual = buildPrettierOptions(editor);
+
+    expect(actual).toEqual({ ...fakePrettierOptions, ...fakeEditorConfigOptions });
+  });
+
+  it('uses settings from prettier-atom and ignores Editor Config when disabled', () => {
+    const editor = buildMockEditor();
+    const fakePrettierOptions = { singleQuote: true, tabWidth: 2 };
+    const mockPrettierInstance = { resolveConfig: { sync: jest.fn(() => null) } };
+    getPrettierInstance.mockImplementation(() => mockPrettierInstance);
+    shouldUseEditorConfig.mockImplementation(() => false);
+    getCurrentFilePath.mockImplementation(() => 'parent-dir/foo.js');
+    getPrettierOptions.mockImplementation(() => fakePrettierOptions);
+
+    const actual = buildPrettierOptions(editor);
+
+    expect(buildEditorConfigOptions).not.toHaveBeenCalled();
+    expect(actual).toEqual(fakePrettierOptions);
+  });
```

Comment:
- In the first test, let's add some line returns, so we have a block each for given, when, and then? This allows the reader to jump straight to the setup, or straight to the action under test, or straight to the expectation.
- Usually, when you test you just write the base case and then specify certain edge cases, that way you don't end up with all of your tests having to test that things **didn't** like crazy.
- This "base case" or "happy path" test should be the first test, and then other tests can just specify how they diverge from the path.

All together:

```js
describe('when a prettier config is found', () => {
  it('returns settings from the prettier config', () => {
    const editor = buildMockEditor();
    const fakePrettierConfigOptions = { tabWidth: 4, printWidth: 100, useTabs: true };
    const mockPrettierInstance = { resolveConfig: { sync: jest.fn(() => fakePrettierConfigOptions) } };
    getPrettierInstance.mockImplementation(() => mockPrettierInstance);
    getCurrentFilePath.mockImplementation(() => 'parent-dir/foo.js');

    const actual = buildPrettierOptions(editor);

    expect(actual).toEqual(fakePrettierConfigOptions);
  });
  
  it('tells prettier whether to use editorconfig when resolving options', () => {
    const editor = buildMockEditor();
    const mockPrettierInstance = { resolveConfig: { sync: jest.fn(() => ({})) } };
    getPrettierInstance.mockImplementation(() => mockPrettierInstance);
    shouldUseEditorConfig.mockImplementation(() => true);
    getCurrentFilePath.mockImplementation(() => 'parent-dir/foo.js');
    
    buildPrettierOptions(editor);
    
    expect(mockPrettierInstance.resolveConfig.sync.mock.calls[0][1]).toEqual({ editorconfig: true });
  });
});

describe('when a prettier config is not found', () => {
  it('returns settings from prettier-atom', () => {
    const editor = buildMockEditor();
    const fakePrettierOptions = { singleQuote: true, tabWidth: 2 };
    const mockPrettierInstance = { resolveConfig: { sync: jest.fn(() => null) } };
    getPrettierInstance.mockImplementation(() => mockPrettierInstance);
    shouldUseEditorConfig.mockImplementation(() => false);
    getCurrentFilePath.mockImplementation(() => 'parent-dir/foo.js');
    getPrettierOptions.mockImplementation(() => fakePrettierOptions);

    const actual = buildPrettierOptions(editor);

    expect(actual).toEqual(fakePrettierOptions);
  });

  it('overrides prettier-atom settings with editorconfig values if editor config is enabled', () => {
    const editor = buildMockEditor();
    const fakePrettierOptions = { singleQuote: true, tabWidth: 2 };
    const fakeEditorConfigOptions = { tabWidth: 4 };
    const mockPrettierInstance = { resolveConfig: { sync: jest.fn(() => null) } };
    getPrettierInstance.mockImplementation(() => mockPrettierInstance);
    buildEditorConfigOptions.mockImplementation(() => fakeEditorConfigOptions);
    shouldUseEditorConfig.mockImplementation(() => true);
    getCurrentFilePath.mockImplementation(() => 'parent-dir/foo.js');
    getPrettierOptions.mockImplementation(() => fakePrettierOptions);

    const actual = buildPrettierOptions(editor);

    expect(actual).toEqual({ ...fakePrettierOptions, ...fakeEditorConfigOptions });
  });
});
```

Reply:
We may be able to utilize prettier for the `overrides prettier-atom settings with editorconfig values if editor config is enabled` case. If you ask for settings from prettier with `editorconfig` enabled, it should return an object where it found the editor config and combined those with prettier defaults for us. No need to implement a custom `editorconfig` solution.

Reply:
@SavePointSam but then we won't know what was from the editorconfig (which we want to keep) and which is just a prettier default (which we want to ignore)?

Reply:
This is the point I was addressing in my comment on the thread below.  Because there's no way for a user to declare a default in the plugin settings, we can't possibly have a 1-to-1 solution matching prettier's behavior.